python/src/rtree.rs

```diff
@@ -103,12 +102,17 @@ pub(crate) enum PyRTreeRef {
 
 impl<'py> FromPyObject<'py> for PyRTreeRef {
     fn extract_bound(ob: &Bound<'py, PyAny>) -> PyResult<Self> {
-        let buffer = PyU8Buffer::extract_bound(ob)?;
-        let ct = CoordType::from_buffer(&buffer.as_ref()).unwrap();
+        let buffer = ob.extract::<PyArrowBuffer>()?;
+        let ct = CoordType::from_buffer(&buffer.as_ref().as_ref()).unwrap();
```

Comment:
This can be reduced to one `as_ref` as of https://github.com/kylebarron/arro3/pull/242